### PR TITLE
1425 url pattern improvement

### PR DESCRIPTION
**Pull request by hyi**
@alvacouch @pkdash This PR needs to be reviewed together with the corresponding PR for django_irods submodule here: https://github.com/hydroshare/django_irods/pull/17. After this PR is merged, federation zone name will be removed from all urls used in HydroShare. So resourcemap.xml download url will be www.hydroshare.org/resource/id/data/resourcemap.xml regardless of which iRODS zone the resource resides. Thanks @alvacouch for pushing forward with this url issue and I am glad I have this issue worked out before federation zone really comes into full use. I am hoping this PR can be merged sooner rather than later since resourcemap.xml file needs to be regenerated for resources that reside in a federated zone. We only have very few such resources as of now, so it'd be good to have this url issue fixed before the number of such resources grow over time.

### CODE REVIEW

File: hs_core/hydroshare/hs_bagit.py [hunk 1]
```diff
@@ -154,26 +149,22 @@ def create_bag_files(resource, fed_zone_home_path=''):
             # in federated zone
             from_fname = f.fed_resource_file_name_or_path
             filename = from_fname.rsplit('/')[-1]
-            res_path = os.path.join(
-                '{hs_url}/resource{zone}/{res_id}/data/contents/{file_name}'.format(
+            res_path = '{hs_url}/resource/{res_id}/data/contents/{file_name}'.format(
```
Comment (pkdash) [inline hs_core/hydroshare/hs_bagit.py]: Would be good avoid this duplicate code.

**Comment by pkdash** [comment]
@hyi You may want to resolve the conflicts with djnago_irords.

File: hs_core/hydroshare/hs_bagit.py (hunk 1 above)
Comment (hyi) [inline hs_core/hydroshare/hs_bagit.py]: @pkdash Done

**Comment by hyi** [comment]
@pkdash I cannot change SHA reference for django_irods in this PR since it points to a branch of django_irods which has not been merged yet. The conflict from django_irods resulted from a recent change to django_irods to fix the issue #1467. Once my django_irods PR for this issue gets merged to master branch of django_irods, I will change SHA reference to master of django_irods to fix this conflict before merging this branch. I cannot do so until django_irods PR is merged into master.

**Comment by hyi** [comment]
@pkdash I have addressed your review comments and see if you can +1 now.
@alvacouch Hope you can code review this sometime today.

File: hs_core/hydroshare/hs_bagit.py [hunk 2]
```diff
@@ -107,13 +107,8 @@ def create_bag_files(resource, fed_zone_home_path=''):
...
     # make the resource map
     current_site_url = current_site_url()
-    if fed_zone_home_path:
-        hs_res_url = '{hs_url}/resource{zone}/{res_id}/data'.format(hs_url=current_site_url,
-                                                                    zone=fed_zone_home_path,
-                                                                    res_id=resource.short_id)
-    else:
-        hs_res_url = '{hs_url}/resource/{res_id}/data'.format(hs_url=current_site_url,
-                                                              res_id=resource.short_id)
+    hs_res_url = '{hs_url}/resource/{res_id}/data'.format(hs_url=current_site_url,
```
Comment (alvacouch) [inline hs_core/hydroshare/hs_bagit.py]: This is good.

Comment (alvacouch) [inline hs_core/hydroshare/hs_bagit.py]: All changes in this vein look great. +1

**Pull request merged**